modules/contact/src/actions/ContactAction.C

```diff
@@ -179,6 +186,105 @@ ContactAction::act()
         addDiracContact();
     }
   }
+
+  if (_current_task == "add_aux_kernel")
+  { // Add ContactPenetrationAuxAction.
+    if (!_problem->getDisplacedProblem())
+    {
+      mooseError("Contact requires updated coordinates.  Use the 'displacements = ...' line in the "
+                 "Mesh block.");
+    }
+
+    {
+      InputParameters params = _factory.getValidParams("PenetrationAux");
+      params.set<ExecFlagEnum>("execute_on") = {EXEC_INITIAL, EXEC_LINEAR};
+
+      // Extract global params
+      if (isParamValid("parser_syntax"))
+        _app.parser().extractParams(getParam<std::string>("parser_syntax"), params);
```

Comment:
These types of action methods are scary...can't you apply params directly through applySpecificParams()?

Reply:
I cannot find a straightforward way to avoid that line without deeper modifications. There was some discussion in #9851 that points to using `MooseObjectAction` to avoid that call. The added functionality may require changing input files and some extra work.

@bwspenc

Reply:
@tophmatthews It does seem we can avoid that call through `applyParameters(parameters())` without changing the action class type.

@bwspenc ping.